dask/array/tests/test_reductions.py

```diff
@@ -15,6 +17,23 @@ def assert_eq(a, b):
     _assert_eq(a, b, equal_nan=True)
 
 
+@pytest.mark.parametrize('dtype', ['f4', 'i4'])
+@pytest.mark.parametrize('keepdims', [True, False])
+def test_numel(dtype, keepdims):
+    x = np.ones((2, 3, 4))
+
+    for length in range(x.ndim):
+        for sub in itertools.combinations([x for x in range(x.ndim)], length):
```

Comment:
So it turns out that Python 2.7 was overwriting the `x` array created before with the `x` from this list comprehension. Do you know which one is the "correct" behavior, @mrocklin? I'm supposing Python standard defines this somewhere, and perhaps this changed in Python 3.

Reply:
Yeah, I think that Python 3 handled scoping issues like this a bit better.  Temporary variables within a comprehension no longer overwrite variables in the outer scope.

Reply:
Thanks for the info @mrocklin.